ara/models.py

```diff
@@ -12,47 +12,105 @@
 #   License for the specific language governing permissions and limitations
 #   under the License.
 
+from datetime import datetime
+import uuid
 from ara import db
 
 
-class Playbooks(db.Model):
-    id = db.Column(db.String, primary_key=True, nullable=False)
-    playbook = db.Column(db.String)
-    start = db.Column(db.String)
-    end = db.Column(db.String)
-    duration = db.Column(db.String)
-    tasks = db.relationship('Tasks', backref='playbooks', lazy='dynamic')
-    stats = db.relationship('Stats', backref='playbooks', lazy='dynamic')
+class TimedEntity(object):
+    @property
+    def duration(self):
+        return self.end - self.start
+
+    def start(self):
+        self.time_start = datetime.now()
+
+    def stop(self):
+        self.time_end = datetime.now()
+
+
+class Playbook(db.Model, TimedEntity):
+    __tablename__ = 'playbooks'
+
+    id = db.Column(db.String(36), primary_key=True, nullable=False,
+                   default=uuid.uuid4)
+    path = db.Column(db.Text)
+    plays = db.relationship('Play', backref='playbook', lazy='dynamic')
+    tasks = db.relationship('Task', backref='playbook', lazy='dynamic')
+    stats = db.relationship('Stats', backref='playbook', lazy='dynamic')
+
+    time_start = db.Column(db.DateTime, default=datetime.now)
+    time_end = db.Column(db.DateTime)
 
     def __repr__(self):
-        return '<Playbook %r>' % self.playbook
-
-
-class Tasks(db.Model):
-    id = db.Column(db.String, primary_key=True, nullable=False)
-    playbook_uuid = db.Column(db.String, db.ForeignKey('playbooks.id'))
-    host = db.Column(db.String)
-    play = db.Column(db.String)
-    task = db.Column(db.String)
-    module = db.Column(db.String)
-    start = db.Column(db.String)
-    end = db.Column(db.String)
-    duration = db.Column(db.String)
+        return '<Playbook %r>' % self.path
+
+
+class Play(db.Model, TimedEntity):
+    __tablename__ = 'plays'
+
+    id = db.Column(db.String(36), primary_key=True, nullable=False,
+                   default=uuid.uuid4)
+    playbook_id = db.Column(db.String(36), db.ForeignKey('playbooks.id'))
+    name = db.Column(db.Text)
+    tasks = db.relationship('Task', backref='play', lazy='dynamic')
+
+    time_start = db.Column(db.DateTime, default=datetime.now)
```

Comment:
I'm not following you.

The record is inserted at the beginning of the playbook/play/task, so having a default here seems fine.

On the other hand, `time_end` always needs to be set explicitly.